daemon/daemon_unix.go

```diff
@@ -950,30 +949,37 @@ func initBridgeDriver(controller *libnetwork.Controller, config *config.Config)
 
 	ipamV4Conf := &libnetwork.IpamConf{AuxAddresses: make(map[string]string)}
 
-	nwList, nw6List, err := netutils.ElectInterfaceAddresses(bridgeName)
+	// By default, libnetwork will request an arbitrary available address
+	// pool for the network from the configured IPAM allocator.
+	// Configure it to use the IPv4 network ranges of the existing bridge
+	// interface if one exists with IPv4 addresses assigned to it.
+
+	nwList, nw6List, err := ifaceAddrs(bridgeName)
 	if err != nil {
 		return errors.Wrap(err, "list bridge addresses failed")
 	}
 
-	nw := nwList[0]
-	if len(nwList) > 1 && config.BridgeConfig.FixedCIDR != "" {
-		_, fCIDR, err := net.ParseCIDR(config.BridgeConfig.FixedCIDR)
-		if err != nil {
-			return errors.Wrap(err, "parse CIDR failed")
-		}
-		// Iterate through in case there are multiple addresses for the bridge
-		for _, entry := range nwList {
-			if fCIDR.Contains(entry.IP) {
-				nw = entry
-				break
+	if len(nwList) > 0 {
```

Comment:
- I notice that for IPv4, there's now this check added; for IPv6 I don't see a similar check; should there be?
- Are situations where `len(nwList) == 0` (or both `nwList` and `nw6List` being empty) an error condition?

Reply:
> * I notice that for IPv4, there's now this check added; for IPv6 I don't see a similar check; should there be?

No. The `nw6List` returned from `ifaceAddrs` is identical to the value returned from `ElectInterfaceAddresses` in all cases, so nothing changes for IPv6.

> * Are situations where `len(nwList) == 0` (or both `nwList` and `nw6List` being empty) an error condition?

No, as the comment block preceding this code explains.